clinical/src/main/kotlin/com/hartwig/actin/clinical/feed/emc/questionnaire/QuestionnaireExtraction.kt

```diff
@@ -11,9 +11,20 @@ object QuestionnaireExtraction {
     private const val ACTIN_QUESTIONNAIRE_KEYWORD = "ACTIN Questionnaire"
     private const val ACTIVE_LINE_OFFSET = 1
 
-    fun extract(entry: QuestionnaireEntry?): Pair<Questionnaire?, List<QuestionnaireCurationError>> {
+    fun extract(entryList: List<QuestionnaireEntry>): Pair<Questionnaire?, List<QuestionnaireCurationError>> {
+        val extracted = entryList.sortedByDescending(QuestionnaireEntry::authored).firstNotNullOfOrNull { it ->
+            extractQuestionnaire(it)
+        }
+        return if (extracted != null) {
+            extracted
+        } else {
+            null to emptyList()
+        }
```

Comment:
You can use the elvis operator to handle nulls specifically. I would also use `asSequence` for lazy evaluation (https://typealias.com/guides/when-to-use-sequences/):
```suggestion
        return entryList.asSequence().sortedByDescending(QuestionnaireEntry::authored)
            .firstNotNullOfOrNull(::extractQuestionnaire)
            ?: null to emptyList()
```